File: 第四次練習/物件組合/Student.cpp
#include <iostream>
#include <string>
#include "Date.h"
#include "Student.h"
using namespace std;

Student::Student(string newName, Date newDate, int newScore)
{
	name = newName;
	birthDay.setDate(newDate.getMonth(), newDate.getDay(), newDate.getYear());
	score = newScore;
}

void Student::setName(string newN)
{
	name = newN;
}

void Student::setDate(Date newD)
{
	birthDay.setDate(newD.getMonth(), newD.getDay(), newD.getYear());
}

void Student::setScore(int newS)
{
	score = newS;
}

string Student::getName() const
{
	return name;
}

Date Student::getDate() const
{
	return birthDay;
}

int Student::getScore() const
{
	return score;
}

void Student::print()
{
	cout << name << " ";
	cout << birthDay.getMonth() << "/" << birthDay.getDay() << "/" << birthDay.getYear() << " ";
	cout << score << endl;
}

File: python練習/StudentCard_Composition/Student.py
#from Date import Date

class Student:
    def __init__(self, n, d, s):
        self.__name = n
        self.__day = d
        self.__score = s
        
    def setName(self, newName):
        self.__name = newName
        
    def setDate(self, d):
        self.__day = d
        
    def setScore(self, newScore):
        self.__score = newScore
        
    def getName(self):
        return self.__name
    
    def getDate(self):
        return self.__day
    
    def getScore(self):
        return self.__score
    
    def toString(self):
        print(self.__name, end=' ')
        self.__day.toString()
        print(self.__score)